source/FreeRTOS_TCP_Transmission_IPV4.c

```diff
@@ -102,13 +102,6 @@ void prvTCPReturnPacket_IPV4( FreeRTOS_Socket_t * pxSocket,
 
     do
     {
-        /* Use do/while to be able to break out of the flow */
-        if( ( pxNetworkBuffer == NULL ) && ( pxSocket == NULL ) )
```

Comment:
I understand that it's not necessary to check network buffer. But why don't we check socket handler?

Reply:
This function is called only when atleast one of them is not NULL. And both socket null or buffer null cases are handled in this function. Hence it is removed.